sound/soc/sof/topology.c

```diff
@@ -3626,6 +3627,15 @@ static void sof_complete(struct snd_soc_component *scomp)
 	 * IPC. It may be overwritten by alsa-mixer after booting up
 	 */
 	snd_sof_cache_kcontrol_val(scomp);
+
+	/*
+	 * Set up routes for all PCM streams. This step is required to identify the widgets
+	 * and their connections for each PCM stream and save them. This information will be
+	 * used when pipelines are set up/taken down dynamically during PCM open/close.
+	 */
+	ret = sof_pcm_route_set_up_all(sdev);
+	if (ret < 0)
+		dev_warn(sdev->dev, "failed to set up route for all PCM streams\n");
```

Comment:
what happens in this case then? We're left with partial widget lists? If they're optional and we can continue without them, we probably should clean up the parts we did manage to allocate. If they're compulsory, we should fail topology loading?

Reply:
@lyakh it is unclear to me how we should handle this right now. Basically, if we go to the point where we want to set up the routes, then obviously topology loading was complete succesfully. But we'd run into problems starting a PCM because we didnt quite manage to identify the route and set up the widgets. Let me experiment a bit and see what happens